### ISSUE TITLE
Duplicated GPU?

### ISSUE THREAD

**Issue by higi** [RainbowMiner/RainbowMiner#531]
Hello, 

i have two GPU in my PC.
- Radeon VII
- RX580

my problem is RainbowMiner thinks there are 4 GPU instead of two (atleast i think that)

it looks like this "All devices"
  | GPU#00 | AMD | RX580 | 4 GB | 36 | - | -
  | GPU#01 | AMD | gfx906 | 15.81 GB | 60 | - | -
  | GPU#02 | AMD | Ellesmere | 4 GB | 36 | - | -
  | GPU#03 | AMD | gfx906 | 15.81 GB | 60

when i start mining it starts two miners with mixed GPUs GPU#00 + , GPU#00
and so on.

Is this ok?

**Comment by higi** [comment]
second problem: now it starts only one 1 gpu
Finished waiting - starting next run
Get-Process : Cannot find a process with the process identifier 7764.
At C:\Users\higi\Desktop\RainbowMinerbkp\Include.psm1:1556 char:24
+             $Process = Get-Process -Id $_
+                        ~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : ObjectNotFound: (7764:Int32) [Get-Process], ProcessCommandException
+ FullyQualifiedErrorId : NoProcessFoundForGivenId,Microsoft.PowerShell.Commands.GetProcessCommand

**Comment by RainbowMiner** [comment]
Wow, that's weird! No, it should detect two GPU, only. Could you please start "GPUtest.bat" and send me the resulting gputestresults.txt?

**Comment by higi** [comment]
I think it started this way. Instaled Radeon 19.4.3 after that xmr miners (opencl) wont work, so i googled a bit, then i installed 18.4.x without clean instalation. Then i instaled back Radeon 19.4.3.

[gputestresult.txt](https://github.com/RainbowMiner/RainbowMiner/files/3142108/gputestresult.txt)

**Comment by RainbowMiner** [comment]
Yes, it looks like the OpenCL installation(s) on your PC really return two AMD platforms, with two GPUs in each of it. This results in the duplicate GPUs. 
So, it's basically a flawed driver installation. But since your GPU run on both platforms, I will quickly add some extra checks to RainbowMiner's device detection, so that only the first platform will be used.

**Comment by RainbowMiner** [comment]
Would you mind and run another test?
Please download this zip and extract it to your RainbowMiner folder:
[GPUtest.zip](https://github.com/RainbowMiner/RainbowMiner/files/3142533/GPUtest.zip)

.. now run GPUtest.bat again and upload the result file.

**Comment by RainbowMiner** [comment]
If you have already updated to RainbowMiner v4.2.0.4, you will not need to download and unzip GPUtest.zip. It's already included.

**Comment by higi** [comment]
it is working now, THX :)

**Comment by RainbowMiner** [comment]
Perfect! :)

You would do me a great favor, if you would now run GPUtest.bat one last time, and sent me the gputesresults.txt.
